### PR TITLE
build: Initial CMake infrastructure

### PR DESCRIPTION
**Pull request by rleigh-codelibre**
Jenkins Windows workspace layout:

- `source`: this source repo
- `build`: built code
- `install`: installed programs in the bin directory

In the jenkins script:

- set CMAKE_PREFIX_PATH=/path/to/ome-files-bundle
- add /path/to/ome-files-bundle/bin to PATH
- run `source/scripts/jenkins-build`

The jenkins-build script is repurposed from the superbuild, but with all configurable options stripped out, as well as the caching.  All options are set directly in the script.  You'll need to download http://downloads.openmicroscopy.org/ome-files-cpp/0.2.3/18/binaries/ome-files-bundle-0.2.3-VC14-x64-Release-b18.zip and unpack somewhere on `D:`; I suggest we create a top-level directory for performance testing programs and data, and unpack it under there.

### CODE REVIEW

File: scripts/jenkins-build.bat [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+REM Jenkins CI infrastructure, though it may be useful as an example for
+REM how one might use various cmake options.
+
+REM So that statements like "set" work inside nested conditionals
+setlocal ENABLEDELAYEDEXPANSION
+setlocal ENABLEEXTENSIONS
+
+set "script=%0"
+
+set "workspace=%~dp0\.."
```
Comment (sbesson) [inline scripts/jenkins-build.bat]: Using the layout from this PR description (`source/scripts/jenkins-build.bat` under the Jenkins workspace), this set of calls does not detect the correct workspace - see https://ci.openmicroscopy.org/view/Files/job/OME-FILES-performance/6/console.

Looking at the corresponding [ome-cmake-superbuild script](https://github.com/ome/ome-cmake-superbuild/blob/master/scripts/jenkins-build.bat), the difference is that the CI job uses the `-w %WORKSPACE%` to override this logic and pass the workspace environment variable.

**Review by sbesson — CHANGES_REQUESTED** [review]

File: scripts/jenkins-build.bat (hunk 1 above)
Comment (rleigh-codelibre) [reply scripts/jenkins-build.bat]: We can fix this up once the jenkins job is working.  So long as it invokes the script, we can use WORKSPACE from the environment.

Comment (sbesson) [reply scripts/jenkins-build.bat]: Sure, is there something extra that needs to be added to the Jenkins job or are you planning to push an extra commit?

Comment (rleigh-codelibre) [reply scripts/jenkins-build.bat]: I'll add to this PR with another commit.  So long as the job runs `source\scripts\jenkins-build` in a windows cmd shell that should be sufficient.

**Comment by rleigh-codelibre** [comment]
OK, green in https://ci.openmicroscopy.org/view/Experimental/job/OME-FILES-performance/16/

Please could you add workspace cleaning to the job?  The `build` and `install` dirs need to be wiped.

**Review by sbesson — APPROVED** [review]
See https://ci.openmicroscopy.org/view/Experimental/job/OME-FILES-performance/15/ with the workspace cleanup at the beginning of the build. All works as expected. I propose we get this merged and start tackling the benchmarking in a separate PR?

**Comment by rleigh-codelibre** [comment]
Yes, that's all fine with me, thanks.

**Pull request merged**